### ISSUE TITLE
KSVM Kernel `Current Predict` Always nan

### ISSUE THREAD

**Issue by sophie-haynes** [JohnLangford/vowpal_wabbit#1555]
#### Issue:
I've been trying to train VW on some text data, which has been working perfectly using `--loss_function logistic` and `hinge`, but when I try using `--ksvm` instead, it always returns nan for predictions. When `--binary` is included, it turns *all* predictions to -1, whereas when `--link=glf1`  is used, nan is once again retuned. The issue persists across kernels (rbf, linear). 

I've tried looking into previous issues and checking the docs, but I can't find any solutions. I may be doing something stupid, it's just peculiar that it seems to work fine with the loss function param (if I'm being stupid, I apologise in advance!).

I've included my steps below, and created a simplified train/test vw file. Thanks!

#### Environment: 
* macOS High Sierra 10.13.6
* vowpal-wabbit 8.6.1 (problem present in 8.5.0 also)

##### trial.train.vw 
```
-1 | text = no
-1 | text = no
1 | text = yes
1 | text = yes
-1 | text = no
```
##### Train Output: 
```
Lambda = 0
Kernel = linear
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = trial.train.vw
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.000000 0.000000            1            1.0  -1.0000      nan        2
0.000000 0.000000            2            2.0  -1.0000      nan        2
0.000000 0.000000            4            4.0   1.0000      nan        2
0.000000 0.000000            8            8.0   1.0000      nan        2

finished run
number of examples = 12
weighted example sum = 12.000000
weighted label sum = 0.000000
average loss = 0.000000
best constant = 0.000000
best constant's loss = 1.000000
total feature number = 24
Num support = 0
Number of kernel evaluations = 0 Number of cache queries = 0
Total loss = 0.000000
Done freeing model
Done freeing kernel params
Done with finish 
```
##### Test Output: 
```
predictions = trial.preds.csv
Lambda = 0
Kernel = linear
Num weight bits = 18
learning rate = 0.5
initial_t = 0
power_t = 0.5
using no cache
Reading datafile = trial.train.vw
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.000000 0.000000            1            1.0  -1.0000      nan        2
0.000000 0.000000            2            2.0  -1.0000      nan        2
0.000000 0.000000            4            4.0   1.0000      nan        2
0.000000 0.000000            8            8.0   1.0000      nan        2

finished run
number of examples = 12
weighted example sum = 12.000000
weighted label sum = 0.000000
average loss = 0.000000
best constant = 0.000000
best constant's loss = 1.000000
total feature number = 24
Num support = 0
Number of kernel evaluations = 0 Number of cache queries = 0
Total loss = 0.000000
Done freeing model
Done freeing kernel params
Done with finish 
```

**Comment by JohnLangford** [comment]
Thanks.   @alekh can you take a look at this?

**Comment by alekh** [comment]
Sent a pull request. There was some weirdness due to a named lambda parameter in ksvm code for ell_2 regularization, but the actual parameter being set from --l2. I also improved the default value from 0 which was causing nans

**Comment by JohnLangford** [comment]
Thanks, merged :-)
